File: ContactManagerApp/app/src/main/java/com/example/contactmanagerapp/ContactDAO.java
package com.example.contactmanagerapp;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface ContactDAO {

    @Delete
    void deleteContact(Contacts contacts);

    @Insert
    void insert(Contacts contacts);


    @Query("SELECT * FROM contacts_table")
    List<Contacts> getAllContacts();


}

File: kotlin/Contact/app/src/main/java/com/example/contact/room/ContactDAO.kt
package com.example.contact.room

import androidx.lifecycle.LiveData
import androidx.room.*

interface ContactDAO {

    @Insert
    suspend fun insertContact(contact: Contact): Long

    @Update
    suspend fun updateContact(contact: Contact)

    @Delete
    suspend fun deleteContact(contact: Contact)

    @Query("DELETE FROM contact_table")
    suspend fun deleteAll()

    @Query("SELECT * FROM contact_table")
    fun getAllContact(): LiveData<List<Contact>>

}